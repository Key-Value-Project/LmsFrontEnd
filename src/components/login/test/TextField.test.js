/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import LoginTextField from '../TextField';

describe('Check Login TextField', () => {
    test('Rendered', () => {
        const { getByTestId } = render(<LoginTextField />);
        const textField = getByTestId('test-textfield-login');
        expect(textField).toBeTruthy();
    });

    test('Label On TextField Displayed', () => {
        const { getByTestId } = render(<LoginTextField label="User Name" />);
        const textField = getByTestId('test-textfield-login');
        expect(textField.textContent).toContain('User Name');
    });

    test('TextField onChange Trigger', () => {
        const onUserNameChange = jest.fn();
        const { getByTestId } = render(<LoginTextField onUserNameChange={onUserNameChange} />);
        const textField = getByTestId('test-login-input');

        expect(textField).toBeTruthy();
        fireEvent.change(textField, { target: { value: 'test1' } });
        expect(onUserNameChange).toHaveBeenCalled();
        expect(onUserNameChange).toHaveBeenCalledWith('test1');
        expect(textField.value).toBe('test1');
    });

    test('TextField Value Displayed', () => {
        const { getByTestId } = render(<LoginTextField userName="test2" />);
        const textField = getByTestId('test-login-input');
        expect(textField.value).toBe('test2');
    });

    test('snapshot matching', () => {
        const { asFragment } = render(<LoginTextField label="User Name" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
