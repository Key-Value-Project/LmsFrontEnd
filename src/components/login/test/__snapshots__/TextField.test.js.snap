// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Login TextField snapshot matching 1`] = `
<DocumentFragment>
  <span
    data-testid="test-textfield-login"
  >
    <label
      for="User Name"
    >
      User Name
    </label>
    <input
      data-testid="test-login-input"
      value=""
    />
  </span>
</DocumentFragment>
`;
